native/exe/SageLauncher/launcher.cpp

```diff
@@ -993,63 +1014,70 @@ int launchJVMSage(LPSTR lpszCmdLine, HWND hWnd, BOOL bClient, BOOL bService)
 	
 	if (createJavaVM==NULL)
 	{
-		errorMsg("Could not execute jvm.dll.\nPlease install Java Runtime Environment 1.7 or greater", "Java Missing", hWnd);
+		errorMsg("Could not execute jvm.dll.\nPlease install Java Runtime Environment 1.7 or higher (Java 1.8 preferred)", "Java Missing", hWnd);
 		return FALSE;
 	}
 
-	LPTSTR prefFile = new TCHAR[512];
-	LPTSTR jarPath = new TCHAR[4096];
-	LPTSTR libraryPath = new TCHAR[512];
+	size_t sizeprefFile = 512;
+	LPTSTR prefFile = new TCHAR[sizeprefFile];
+	size_t sizejarPath = 4096;
+	LPTSTR jarPath = new TCHAR[sizejarPath];
+	size_t sizelibraryPath = 512;
+	LPTSTR libraryPath = new TCHAR[sizelibraryPath];
 	
 	stdOutHandle = GetStdHandle(STD_OUTPUT_HANDLE);
     JNIEnv *env;       /* pointer to native method interface */
 	JavaVMInitArgs vm_args;
 	JavaVMOption options[32];
 	vm_args.nOptions = 0;
 #ifdef SAGE_TV
-	strcpy(prefFile, bClient ? "sagetvclient " : "sagetv ");
-	strcat(prefFile, appPath);
-	strcat(prefFile, bClient ? "SageClient.properties" : "Sage.properties");
-	strcpy(jarPath, "-Djava.class.path=");
-	strcat(jarPath, appPath);
-	strcat(jarPath, "Sage.jar;");
-  strcat(jarPath, appPath);
-  strcat(jarPath, "JARs\\lucene-core-3.6.0.jar;");
-	strcat(jarPath, appPath);
-	strcat(jarPath, "xerces.jar;");
-	strcat(jarPath, appPath);
-	strcat(jarPath, "plugin.jar;");
-	strcat(jarPath, appPath);
-	strcat(jarPath, ";");
-	char* envcp = getenv("CLASSPATH");
+	strcpy_s(prefFile, sizeprefFile, bClient ? "sagetvclient " : "sagetv ");

```

Comment:
With the stack allocations above, then you can just use sizeof in all these calls to make it cleaner.